ibtk/include/ibtk/HierarchyIntegrator.h

```diff
@@ -493,6 +493,24 @@ class HierarchyIntegrator : public SAMRAI::mesh::StandardTagAndInitStrategy<NDIM
      */
     void registerApplyGradientDetectorCallback(ApplyGradientDetectorCallbackFcnPtr callback, void* ctx = nullptr);
 
+    /*!
+     * Callback function specification to enable further specialization of regridHierarchy().
+     */
+    using RegridHierarchyCallbackFcnPtr =
+        void (*)(SAMRAI::tbox::Pointer<SAMRAI::hier::BasePatchHierarchy<NDIM> > hierarchy,
+                 double data_time,
+                 bool initial_time,
+                 void* ctx);
+
+    /*!
+     * Register a callback function to enable further specialization of regridHierarchy().
+     *
+     * \note This function will be called after all data owned by the hierarchy is initialized and set, but before the
+     * data is synchronized. If data owned by the integrator is used in this function, it should be synchronized prior
+     * to use.
```

Comment:
From the other conversation: if this is called prior to synchronization but a user needs synchronized data, then how should they do it?

Reply:
Using an appropriate coarsening operation. I've updated the comment.